Add MESMER_M Changelog entries (#463)
MESMER-group/mesmer: CHANGELOG.rst

- move MESMER-M scripts into mesmer, adapt init file and small issues (
  `#419 <https://github.com/MESMER-group/mesmer/pull/419>`_,
  `#421 <https://github.com/MESMER-group/mesmer/pull/421>`_).
- documentation (
  `#424 <https://github.com/MESMER-group/mesmer/pull/424>`_,
  `#425 <https://github.com/MESMER-group/mesmer/pull/425>`_)
- rewriting power transformer functions to work on xarrays, this also includes a move from a class structure to functions (
  `#442 <https://github.com/MESMER-group/mesmer/pull/442>`_)
- refactoring of MESMER-M functions (
  `#415 <https://github.com/MESMER-group/mesmer/pull/415>`_,
  `#434 <https://github.com/MESMER-group/mesmer/pull/434>`_,
  `#439 <https://github.com/MESMER-group/mesmer/pull/439>`_)
- add utility functions for MESMER-M (
  `#418 <https://github.com/MESMER-group/mesmer/pull/418>`_,
  `#435 <https://github.com/MESMER-group/mesmer/pull/435>`_)
- adjust the first guess for the power transformer to 0 and 1 (`#429 <https://github.com/MESMER-group/mesmer/pull/429/files>`_)
- adjust the bounds for the power transformer to all positive numbers for first coefficient (`#446 <https://github.com/MESMER-group/mesmer/pull/446>`_))

> - optimizing (

Reviewer: ```suggestion
- performance optimization (
```

- Unshallow the mesmer git repository on rtd (`#456 <https://github.com/MESMER-group/mesmer/pull/456>`_).
  By `Victoria Bauer`_.

Integration of MESMER-M
^^^^^^^^^^^^^^^^^^^^^^^

This version integrates MESMER-M into the remaining MESMER infrastructure. This includes

> some refactoring, bugfixes and enhancements of the MESMER-M functionality.
> 

Reviewer: ```suggestion
some refactoring, bugfixes and enhancements of the MESMER-M functionality. Note
that this led to some numerical changes compared to the MESMER-M publication
(Nath et al., (2022)[https://doi.org/10.5194/esd-13-851-2022]).

```

- Unshallow the mesmer git repository on rtd (`#456 <https://github.com/MESMER-group/mesmer/pull/456>`_).
  By `Victoria Bauer`_.

Integration of MESMER-M
^^^^^^^^^^^^^^^^^^^^^^^

> This version integrates MESMER-M into the remaining MESMER infrastructure. This includes

Reviewer: ```suggestion
This release integrates MESMER-M into the existing MESMER infrastructure. This includes
```

some refactoring, bugfixes and enhancements of the MESMER-M functionality. Note
that this led to some numerical changes compared to the MESMER-M publication

> (Nath et al., (2022)[https://doi.org/10.5194/esd-13-851-2022]).

Reviewer: ```suggestion
(Nath et al., [2022](https://doi.org/10.5194/esd-13-851-2022)).
```

:man_shrugging:
